### ISSUE TITLE
Failing to parse OpenAPI into valid hydra_doc

### ISSUE THREAD

**Issue by gustavodemorais** [HTTP-APIs/hydra-openapi-parser#14]
### I'm submitting a 
- [x] bug report.
- [ ] feature request.

### Current Behaviour:

Currently the parse method does not provide a valid output for the "petstore_openapi.yaml". It creates invalid expects parameters for some Endpoints:
![untitled (4)](https://user-images.githubusercontent.com/11739405/54075314-3308ee00-429e-11e9-98ac-39b6fe4cf218.png)

Therefore, when dumping the hydra_doc it fails:
![image](https://user-images.githubusercontent.com/11739405/54075321-4916ae80-429e-11e9-8b44-e7cdfa9fbec9.png)

### Expected Behaviour:

For the file to be parsed and successfully create a valid "hydra_doc_samply.py" under the samples folder. 

### Steps to reproduce:

    git clone https://github.com/HTTP-APIs/hydra-openapi-parser
    cd hydra-openapi-parser
    python3 -m venv venv
    source venv/bin/activate
    pip install -r requirements.txt 
    python setup.py install
    cd hydra_openapi_parser
    python openapi_parser.py

### Do you want to work on this issue?

Yes, I'm trying to fit the Open API Parser as part of my GSoC proposal.

**Comment by gustavodemorais** [comment]
I've already taken some steps trying to figure this out:

1. Test currently works because there are no tests for dumping.

2. The last fully working version I found was back in hydrus, commit caabdb1e4b6d910cb4253a869814792e137f0d42

3. I was able to make it work moving forward to the next commit and manually merging check_for_ref and check_collection but still think it's not working 100% properly

Will continue to look into it, suggestions and corrections are appreciated.
